src/agents/agent.py

```diff
@@ -63,7 +64,16 @@ def search_queries(self, queries: list, project_name: str) -> dict:
         results = {}
         
         knowledge_base = KnowledgeBase()
-        bing_search = BingSearch()
+        user_search_engine = Config().get_search_engine()
+
+        if user_search_engine == "DDGS":
+            search_engine = DuckDuckGoSearch()
+        else:
+            """
+            Set default search engine to Bing
+            """
+            search_engine = BingSearch()
+            
```

Comment:
**Performance Issue**: The choice of search engine is hardcoded to either DuckDuckGoSearch or BingSearch based on a configuration setting, but there's no validation to ensure the configuration value is correct. This could lead to a situation where an invalid value causes the application to default to BingSearch without any indication of a misconfiguration to the user. <br> **Fix**: Validate the 'user_search_engine' configuration value against a list of supported search engines. If the value is not supported, log a warning or error and default to BingSearch. Additionally, inform the user of the misconfiguration through appropriate application logging or user interface feedback mechanisms. <br> **Code Suggestion**: 
 ```
 +SUPPORTED_SEARCH_ENGINES = ['Bing', 'DDGS']
 
  class Config:
      def get_search_engine(self):
 +        engine = self.config['USER_PREFS']['SEARCH_ENGINE']
 +        if engine not in SUPPORTED_SEARCH_ENGINES:
 +            log.error(f'Unsupported search engine: {engine}. Defaulting to Bing.')
 +            return 'Bing'
 +        return engine

 ```